app/javascript/src/web-components/save-button.js

```diff
@@ -116,10 +120,10 @@ class SaveButton extends HTMLButtonElement {
 
   // Enables programatically submitting the form, regardless of the state of the
   // button, and bypassing unload prevention.
-  submit() {
+  save() {
     this.#enabled = true;
-    this.#removeBeforeUnloadListener();
-    this.form.submit(); // Does not raise a `submit` event
+    this.form.dispatchEvent(new Event('submit')) // Trigger submit event listeners
```

Comment:
Is this now a behaviour change? Looks like the old code didn't trigger the event? Or did it trigger in `submit()` so now we need to do it manually?

Reply:
Which old code are you referring to? `SubmitHandler`, or the previous `save-button` implementation?

`SubmitHandler` used `$form.submit()` which calls jQuery's `submit()` function, which tirggered event listeners.

We now use `form.submit()` which is the browser native submit method which **does not** trigger event listeners.  (There is a newer `requestSubmit()` method, which calls the validation API and triggers listeners, however it's support only goes back one or two safari versions, which is not ideal so I didn't use it.)

The previous `save-button` implementation wasn't hindered by the event listeners not firing as there was nothing listening for submit events on settings screens.